lib/datadog/tracing/trace_operation.rb

```diff
@@ -404,6 +425,7 @@ def build_trace(spans, partial = false)
           name: @name,
           resource: @resource,
           service: @service,
+          tags: !partial ? @tags : nil,
```

Comment:
Not sure if this is quite right.

Right now I think it's accurate if we're thinking it only wants to be on the root span. (This is the responsibility of the `TraceFormatter` to make this determination though.)

However, when we change trace tags from root span tagging to trace payload, I'm not sure if this becomes incorrect. I think it depends on whether you can submit the same tag for a trace multiple times. If yes, then incorrect, if no, then this becomes very important.

Reply:
> This is the responsibility of the TraceFormatter to make this determination though

The TraceFormatter has no direct knowledge of a TraceSegment representing a partial trace flush, but I think I found a way to achieve that in the TraceFormatter (using `@found_root_span`, as is done in `set_resource!`).

Reply:
Moved logic to TraceFormatter as described.